### PR TITLE
add new types for notification service

### PR DESCRIPTION
**Pull request by Daniel-GrunbergerCA**

### CODE REVIEW

File: armotypes/notificationreporttypes.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
+package armotypes
+
+type WeeklyReport struct {
```
Comment (avrahams) [inline armotypes/notificationreporttypes.go:3]: Please add this type  to NotificationsConfig struct (in portalType.go) as "latestReport" field (pointer,omitempty)

It will also be nice to move NotificationsConfig to this file

File: armotypes/notificationreporttypes.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	RepositoriesScannedThisWeek         int                 `json:"repositoriesScannedThisWeek" bson:"repositoriesScannedThisWeek"`
+	RepositoriesScannedPrevWeek         int                 `json:"repositoriesScannedPrevWeek" bson:"repositoriesScannedPrevWeek"`
+	LinkToRepositoriesScanningFiltered  string              `json:"linkToRepositoriesScanningFiltered" bson:"linkToRepositoriesScanningFiltered"`
+	RegistriesScannedThisWeek           int                 `json:"registriesScannedThisWeek" bson:"registriesScannedThisWeek"`
+	RegistriesScannedPrevWeek           int                 `json:"registriesScannedPrevWeek" bson:"registriesScannedPrevWeek"`
+	LinkToRegistriesScanningFiltered    string              `json:"linkToRegistriesScanningFiltered" bson:"linkToRegistriesScanningFiltered"`
+	Top5FailedControls                  []TopCtrlItem       `json:"top5FailedControls" bson:"top5FailedControls"`
+	Top5FailedCVEs                      []TopVulItem        `json:"top5FailedCVEs" bson:"top5FailedCVEs"`
+	ClustersScanned                     []ClusterScanned    `json:"clustersScanned" bson:"clustersScanned"`
```
Comment (avrahams) [inline armotypes/notificationreporttypes.go:15]: Can't it be just an array of  '''ClusterResourceScanned'''
For what do we need the ClusterScanned type?

Same question about RepositoryScanned  & RegistryScanned

File: armotypes/notificationreporttypes.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+type RegistryScanned struct {
+	Registry ResourceScanned `json:"registry" bson:"registry"`
+}
+
+type RepositoryScanned struct {
+	Repository ResourceScanned `json:"repository" bson:"repository"`
+}
+
+type ClusterResourceScanned struct {
```
Comment (avrahams) [inline armotypes/notificationreporttypes.go:32]: ClusterResourceScanned  can use ResourceScanned as base struct and just add  shortName
(```fullName'``will be ``name ``)

File: armotypes/notificationreporttypes.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+	Description        string                         `json:"description" bson:"description"`
+	Severity           string                         `json:"severity" bson:"severity"`
+	SeverityScore      int                            `json:"severityScore" bson:"severityScore"`
+	Metadata           interface{}                    `json:"metadata" bson:"metadata"`
+	Fixes              VulFixes                       `json:"fixedIn" bson:"fixedIn"`
+	Relevancy          string                         `json:"relevant" bson:"relevant"`
+	UrgentCount        int                            `json:"urgent" bson:"urgent"`
+	NeglectedCount     int                            `json:"neglected" bson:"neglected"`
+	HealthStatus       string                         `json:"healthStatus" bson:"healthStatus"`
+	Categories         VulnerabilityCategory          `json:"categories" bson:"categories"`
```
Comment (avrahams) [inline armotypes/notificationreporttypes.go:63]: Why do we need this VulnerabilityCategory  wrapper and not just put IsRCE bool here?

File: armotypes/notificationreporttypes.go [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+type FixedIn struct {
+	Name    string `json:"name" bson:"name"`
+	ImgTag  string `json:"imageTag" bson:"imageTag"`
+	Version string `json:"version" bson:"version"`
+}
+type TopVulItem struct {
+	Vulnerability   `json:",inline"`
+	WorkloadsCount  int64 `json:"workloadsCount" bson:"workloadsCount"`
```
Comment (avrahams) [inline armotypes/notificationreporttypes.go:80]: Vulnerability     is not used anywhere else besides in here - so why not adding these two fields to Vulnerability  and remove this struct?

File: armotypes/notificationreporttypes.go [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+type TopCtrlCluster struct {
+	Name               string `json:"name" bson:"name"`
+	ResourcesCount     int64  `json:"resourcesCount" bson:"resourcesCount"`
+	ReportGUID         string `json:"reportGUID" bson:"reportGUID"`
+	TopFailedFramework string `json:"topFailedFramework" bson:"topFailedFramework"`
+}
+
+type CommonContainerScanSummaryResult struct {
```
Comment (avrahams) [inline armotypes/notificationreporttypes.go:104]: Is this related to notifications ?

**Review by avrahams — CHANGES_REQUESTED** [review]

File: armotypes/notificationreporttypes.go (hunk 2 above)
Comment (Daniel-GrunbergerCA) [reply armotypes/notificationreporttypes.go:15]: I made an external type for registry and repository so that they won't have the same type, and it would be clearer. But I will change it (same was for cluster then I added full and short name)

File: armotypes/notificationreporttypes.go (hunk 4 above)
Comment (Daniel-GrunbergerCA) [reply armotypes/notificationreporttypes.go:63]: This was taken from the struct which is returned by BE

File: armotypes/notificationreporttypes.go (hunk 6 above)
Comment (Bezbran) [reply armotypes/notificationreporttypes.go:117]: Let's move the container/posture scan structs to dedicated files

File: armotypes/notificationreporttypes.go [hunk 7]
```diff
@@ -0,0 +1,164 @@
+package armotypes
```
Comment (Bezbran) [inline armotypes/notificationreporttypes.go:1]: file name should be **user**notifications...

**Review by Bezbran — CHANGES_REQUESTED** [review]

**Review by Bezbran — APPROVED** [review]
